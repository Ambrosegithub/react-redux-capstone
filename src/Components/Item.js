import React from 'react';
import { Link } from 'react-router-dom';
import './item.css';

const Item = (state) => {
  const {
    id, title, price, image,
  } = state;

  const UNITPRICE = 'USD';
  return (
    <div className="products">
      <div className="contain" key={id}>
        <img
          className="images"
          src={image}
          alt="Nice cool products"
          style={{ width: '200px', height: '200px' }}
        />
        <div className="sub-cont">
          <h3 className="title">{title}</h3>
          <div className="price">{`${UNITPRICE} ${price}`}</div>
          <Link to={`/products/${id}`}>
            <div className="btn">see product Details</div>
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Item;
