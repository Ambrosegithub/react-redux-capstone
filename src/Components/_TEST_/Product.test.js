import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Item from '../Item';
import store from '../../redux/configure';

describe('Test of Profile Components', () => {
  test('Item Components Renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Item />
          </Provider>
        </BrowserRouter>,
      );
    expect(tree).toMatchSnapshot();
  });
});
