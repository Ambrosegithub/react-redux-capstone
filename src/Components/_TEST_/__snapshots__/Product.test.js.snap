// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Profile Components Item Components Renders correctly 1`] = `
<div
  className="products"
>
  <div>
    <h3
      className="title"
    />
    <div
      className="price"
    >
       
      USD undefined
       
    </div>
    <img
      alt="Nice cool products"
      className="images"
      style={
        Object {
          "height": "200px",
          "width": "200px",
        }
      }
    />
    <a
      href="/products/undefined"
      onClick={[Function]}
    >
      <div>
        see project Details
      </div>
    </a>
  </div>
</div>
`;
